### DISCUSSION TITLE
Draft - WebUI 2.5.0 Beta - Status

### DISCUSSION THREAD

**Discussion by hassandraga** [webui-dev/webui#411] [Announcements]
*** **[ ! ] THIS IS A DRAFT - PLEASE SUGGEST UPDATES IF ANY** ***

**Team Planning**: https://github.com/orgs/webui-dev/projects/1/views/1

**New Features**

- In addition to using web browsers as GUI, now WebUI supports WebView as well (Proposed by @nater0000)
- Automatically refresh the UI when any file in the root folder gets changed (Proposed by @nater0000)
- In addition to `int`, `bool`, and `string`, now WebUI support `float` (Proposed by @crystalthoughts)
- Control how to receive UI events, single thread or multi-threads (Proposed by @dgllghr)
- Setup a custom proxy (By @osljw)
- Easy global config using the new `config()` API
- Get the argument counts
- The pre-built archives now come with the HTTPS TLS version

**New APIs**

- `webui_show_wv()` 
- `webui_set_event_blocking()`
- `webui_get_best_browser()`
- `webui_set_proxy()`
- `webui_get_count()`
- `webui_get_float_at()`
- `webui_get_float()`
- `webui_return_float()`
- `webui_set_config()`
- `enum webui_config`
- `webui_open_url()`
- `webui_start_server()`
- `webui_browser_exist()`
- `webui_script_client()`
- `webui_run_client()`
- `webui_close_client()`
- `webui_navigate_client()`
- `webui_show_client()`
- `webui_send_raw_client()`
- _All details and descriptions are in [webui.h](https://github.com/webui-dev/webui/blob/main/include/webui.h)_

**Breaking Changes**

- ...

**Renamed**

- `enum webui_browsers` to `enum webui_browser`
- `enum webui_runtimes` to `enum webui_runtime`
- `enum webui_events` to `enum webui_event`

**Bugs Fix**

- `webui.js` causes infinite-reload when using [SvelteKit](https://kit.svelte.dev/) (Reported by @sensiblearts)
- Missing propre redirect when requesting a folder `/subFolder` to `/subFolder/index.html` (Reported by @nater0000)

**Comment by ttytm** [comment]
Added a new wrapper for this version.

https://github.com/webui-dev/swift-webui

Most of the heavy work is done. There is not much work left to have the full API finished. The basic already run nicely on Linux and Windows. The compiler just has some complains on macos with civetweb. 

Regarding the macos failure, it's available in the wrappers CI run. Input on how to approach it is very welcome.

**Reply by hassandraga** [reply]
Thank you @ttytm, that is a great work 👍

**Reply by ttytm** [reply]
So far, I'm experiencing the best C and C++ interoperability with Swift.
With the editor properly set up, I'm getting type definitions, symbol references, and code completions for C code directly from Swift.

For the wrapper, it was possible to implement the "native" response return type feature today:

```swift
func handleResp(e: Event) -> Int {
	let count: Int = try! e.getArg()
	return count * 2
}
...
win.bind("handleResp", handleResp)
```

Ref.: https://github.com/webui-dev/swift-webui/blob/fb24f7e8d962e5f9f2d34b17ed8b780ca64a4742/Examples/CallSwiftFromJS/main.swift#L75-L78

**Comment by Juff-Ma** [comment]
So... what is the state of this? Like, at all. WebUI seems to have active development going on, yet the last stable release was almost 3 years ago. 

Additionally, beta.3 has been released over a year ago as well. What is going on?

**Reply by AlbertShown** [reply]
The core is stable enough, but there is still one last feature to finish before releasing the stable version: adding more window control for the web browser based window.

The macOS dock icon issue seems to be an OS behavior rather than a WebUI issue.

**Reply by Juff-Ma** [reply]
I see, glad to hear that :)

It just was a bit worrying seeing bug fixes and very highly awaited features getting included in master but just not being released.
